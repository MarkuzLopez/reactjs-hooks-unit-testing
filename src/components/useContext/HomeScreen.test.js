import { mount, shallow } from "enzyme"
import { HomeScreen } from "./HomeScreen"
import { UserContext } from "./UserContext"

describe('Pruebas unitarias HomeScreen', () => {
    
    // variables globales 

    const user= { 
        name: 'Eliza',
        email: 'eliza@garduno.com'
    }
    // mount para que renderice todo el component y no solo uno
    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen />
        </UserContext.Provider>
    )

    test('debe mostrarse correctament con snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
})
