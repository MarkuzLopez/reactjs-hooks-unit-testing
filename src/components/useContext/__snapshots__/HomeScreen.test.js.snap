// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitarias HomeScreen debe mostrarse correctament con snapshot 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre>
    {
   "name": "Eliza",
   "email": "eliza@garduno.com"
}
  </pre>
</div>
`;
